File: Python/_443_String_Compression/solution.py
class Solution:
    def compress(self, chars):
        """
        :type chars: List[str]
        :rtype: int
        """
        i, j = 0, 0
        while (j < len(chars)):
            if j == len(chars) - 1 or chars[j] != chars[j + 1]:
                chars[i] = chars[j]
                i += 1
                j += 1
            else:
                cur = j
                while j < len(chars) and chars[j] == chars[cur]:
                    j += 1
                chars[i] = chars[cur]
                i += 1
                num = str(j - cur)
                for c in num:
                    chars[i] = c
                    i += 1
        return i

File: Java/_443_String_Compression/Solution.java
class Solution {
    public int compress(char[] chars) {
        int i = 0;
        for (int j = 0; j < chars.length;)
        {
            if (j == chars.length - 1 || chars[j] != chars[j + 1])
                chars[i++] = chars[j++];
            else
            {
                int cur = j;
                while (j < chars.length && chars[j] == chars[cur])
                    j++;
                chars[i++] = chars[cur];
                char[] num = Integer.toString(j - cur).toCharArray();
                for (char c : num)
                    chars[i++] = c;
            }
        }
        return i;
    }
}